src/datasources/asset-calibration/constants.ts

```diff
@@ -0,0 +1,84 @@
+import { ComputedDataFields, QueryBuilderOperations } from "core/query-builder.constants";
+import { AssetTypeOptions, BusTypeOptions, QBField } from "./types";
+
+export const AssetCalibrationFields: Record<string, QBField> = {
+    LOCATION: {
+        label: 'Location',
+        dataField: ComputedDataFields.Location,
+        filterOperations: [
+            QueryBuilderOperations.EQUALS.name,
+            QueryBuilderOperations.DOES_NOT_EQUAL.name
+        ],
+    },
+    WORKSPACE: {
+        label: 'Workspace',
+        dataField: 'Workspace',
+        dataType: 'string',
+        filterOperations: [
+            QueryBuilderOperations.EQUALS.name,
+            QueryBuilderOperations.DOES_NOT_EQUAL.name
+        ],
+        lookup: {
+            dataSource: [
+                {
+                    label: '$workspace',
+                    value: '$workspace'
+                }
```

Comment:
It's been a while since I've touched the query builder. Is this giving the user an option to select a $workspace variable?

I don't think we should do this—we shouldn't assume that the user has defined a workspace variable on the dashboard or that it's called "workspace".

Reply:
Agreed, updated the logic for this kind of query. The user can write his variable, and it will be replaced if it exists.